### DISCUSSION TITLE
DIY CC2652P Zigbee Router (with SMA connector + external antenna) by tube0013

### DISCUSSION THREAD

**Discussion by Hedda** [zigpy/zigpy#664] [General]
@tube0013 have posted some info in Home Assistant community about his DIY TI CC2652P based dedicated Zigbee Router project:

https://community.home-assistant.io/t/zigbee-router-on-steroids/280896

https://github.com/tube0013/tube_gateways/tree/main/tube_cc_router

https://www.tubeszb.com/shop/routers/3

_**"I’m going to release the design on my github as well as make them to sell."**_

Looks to be based on CC2652P ([RF-Star CC2652P2](https://www.szrfstar.com/product/237-en.html)) so assume it can also be flashed with Zigbee coordinator FW and used with a USB-cable?

![image](https://user-images.githubusercontent.com/6320001/108036193-34212900-7038-11eb-9891-8889527942c5.png)

![image](https://user-images.githubusercontent.com/6320001/108036332-5e72e680-7038-11eb-910f-2754e4c4db8b.png)

**Comment by Hedda** [comment]
@tube0013 Will the final enclosure have paperclip 'reset' holes so can reach/press “pairing” and" bootloader" buttons without opening it up?

Another niche marketing point could be that the RF-star RF-BM-2652P2 module claims to work at extreme temperaturs, if the user would put both the board and an AC-to-DC power-supply that support similar working tempetures into a ABS waterproof enclosure with IP66 code or better and seal both power-cable and SMA-connector with marine grade silicone sealant then it should be possible to use that as an ourdoor Zigbee router in both rain and snow. 

https://www.itead.cc/sonoff-ip66.html

https://www.itead.cc/waterproof-plastic-case.html

Reason for use-case idea is that I have several [Ledvance/Osram Smart+ Outdoor Plug](https://zigbee.blakadder.com/OSRAM_31930.html)(s) outside today, they have really horrible reception so have been thinking about putting a Zigbee router outside (and we in north-east part of Sweden which has cold climate with extreme weather).

**Comment by MattWestb** [comment]
I was thinking we (ZHA users) was sitting in the famous "first row" (with (@SensorsIot) or without Swiss accent) but we ar **not** doing that  :-(((

As normal with tube very nice design and have picked good components and good format of the PCB and casing.

It would being interesting having one 240 V AC => 5 V DC converter with the same format that can being mounted under the PCB in one 2 chamber casing.

Great @tube0013 ! ! !

**Reply by MattWestb** [reply]
I was forgetting the name.

"_**Tube Pure Blue**_"

**Reply by Hedda** [reply]
Note! CC2652P2 supports ~ 1.8 V - 3.8 V (3.3 V is recommend), not 5 V, and need one for 100-240V AC for global mains.

https://www.szrfstar.com/product/237-en.html

You can buy internal AC-DC converters, ex: Hi-Link HLK-2M03 / HLK-PM03 or better yet Mean Well IRM-02-3.3 / IRM-03-3.3

http://www.hlktech.net/product_detail.php?ProId=59

https://www.meanwell-web.com/en-gb/ac-dc-single-output-encapsulated-power-supply-irm--02--3.3

https://www.meanwell-web.com/en-gb/ac-dc-single-output-encapsulated-power-supply-pcb-irm--03--3.3

https://www.meanwell.com/Upload/PDF/IRM-03/IRM-03-SPEC.PDF

https://docs.rs-online.com/5330/0900766b815c758c.pdf

Zigbee2mqtt has instructions to convert 100-240 volt AC power to 3.3 volt DC for a Zigbee router using HLK-PM03 module:

https://www.zigbee2mqtt.io/how_tos/how_to_create_a_cc2530_router.html

![image](https://user-images.githubusercontent.com/6320001/108056346-6be89a80-7051-11eb-862e-437d9c77e70f.png)

![image](https://user-images.githubusercontent.com/6320001/108227154-e262c680-713d-11eb-9847-70446d0e7141.png)


![image](https://user-images.githubusercontent.com/6320001/108227033-c0694400-713d-11eb-8483-689808dfd0d9.png)


Wonder minimum peak Amps/Watt required for RF-Star CC2652P2 based Zigbee router powered by 110V verses 230V?

PS: Tips is to do a google search for "ESP8266 mains power" to find many guides on safe AC to DC converting. Example:

https://randomnerdtutorials.com/esp8266-hi-link-hlk-pm03/

**Comment by tube0013** [comment]
Thanks Guys!!!

1. There is no serial converter onboard for this to run as a coordinator otb. Headers are there so a serial converter could transform this into a coordinator with additional hardware. Goal was to make it light on components and simple.

2. No pin hole in current design the top is a friction fit and not difficult to remove. Holding the physical button vs holding a pin to it is much easier (for me at least)

3. I'm not venturing into mains power anytime soon ⚡️😬

4. More designs coming, I don't think I'm going to look at an efr32mg equivalent to this router design though, the module availability and options are not abundant on that side.I'll see what happens though.

**Reply by Hedda** [reply]
> I don't think I'm going to look at an efr32mg equivalent to this router design though, the module availability and options are not abundant on that side.

@tube0013  FYI, if you are looking to buy modules as surface mounted components to make and sell boards then know that MGM210P (and MGM210L) RF modules by Silicon Labs that contains EFR32MG21 and are also available.

https://www.silabs.com/wireless/zigbee/efr32mg21-series-2-modules

https://www.silabs.com/documents/public/data-sheets/mgm210l-datasheet.pdf

https://www.silabs.com/documents/public/data-sheets/mgm210p-datasheet.pdf

MGM210PA32JIA and MGM210PB32JIA are both is a +20 dBm TX pre-certified module based module with both built-in antenna and RF Pin for optional external antenna. 

https://www.silabs.com/wireless/zigbee/efr32mg21-series-2-modules/device.mgm210pa32jia

https://www.silabs.com/wireless/zigbee/efr32mg21-series-2-modules/device.mgm210pb32jia

The differnce between those two is that MGM210PB32JIA also have support for features Silicon Labs' enhanced security option, "Secure Vault" https://www.silabs.com/security

MGM210PA32JIA and MGM210PB32JIA looks to cost about $11 (US) dollars each when not buying in bulk

https://octopart.com/click/track?ai=1872&sig=02b5980&sid=459&ppid=108099587&vpid=615228293&ct=offers

https://octopart.com/click/track?ai=1872&sig=045aafa&sid=2401&ppid=108099587&vpid=618359838&ct=offers

https://octopart.com/click/track?c=1&ai=1872&sig=0881c5f&sid=459&ppid=112348737&vpid=667261090&ct=offers

https://octopart.com/click/track?ai=1872&sig=09e2041&sid=819&ppid=112348737&vpid=692111326&ct=offers

**Reply by Hedda** [reply]
> pre-certified module

Believe these are already Zigbee Alliance certified so devices can get listed https://zigbeealliance.org/product_type/certified_product/

**Reply by tube0013** [reply]
I just need this package to arrive... it made it out of China before the New Year. But has been in purgatory since arriving in the US 🤦‍♂️![image](https://user-images.githubusercontent.com/11084412/108077455-87947680-703a-11eb-9103-91e1af4261ae.jpeg)

**Reply by MattWestb** [reply]
Is JFK  in shutdown of do they only not liking you ?
Happy Chinese new year :-))

**Reply by Hedda** [reply]
> No pin hole in current design the top is a friction fit and not difficult to remove. Holding the physical button vs holding a pin to it is much easier (for me at least)

OK but maybe an alternative could otherwise be to make a loose button in the cover like the solution concepts from these designs? 

https://www.thingiverse.com/thing:2706968

https://www.thingiverse.com/thing:2797518

I have no issues with opening it myself however still like the idea that users who buys a product not having to open the enclosure ;)

**Comment by Hedda** [comment]
> There is no serial converter onboard for this to run as a coordinator otb. Headers are there so a serial converter could transform this into a coordinator with additional hardware. Goal was to make it light on components and simple.

Would it not also be a huge risk to not include an onboard USB to Serial converter as then firmware can only be OTA updateded?

How much extra cost would it be in parts + work to include a onboard USB to Serial/UART Bridge chip to the final board design?

Keeping in mind that should expect that users buying one from you might want to upgrade the firmware on it sooner or later.

You might by the way be interested in reading discussion about such chips (and more) in https://github.com/electrolama/zig-a-zig-ah/issues/12 regardless.

puddly recommends USB-to-serial chip by FTDI or Silicon Labs (CP210x series) as CH340 by Texas Instruments been unreliable.

**Comment by Hedda** [comment]
Posted this question in the HA community forum but might as well post it here for reference to other creators/makers/builders:

@tube0013 Have you considered buying a budget-friendly resin 3D-printer with a relatively large build volume such as; Anycubic Photon Mono X, Elegoo Saturn, Epax E10, Phrozen Sonic Mighty 4K, Peopoly Phenom, Phrozen Transform, or similar? 

...as well as a resin washing and UV curing station of course ;)

Tip is to read this article just to become aware that using large resin 3D-printers is now an option for small businesses on a budget:

https://all3dp.com/1/best-large-format-resin-dlp-sla-3d-printer-stereolithography/

> Resin 3D printer popularity is surging now that the technology is no longer confined to professional machines costing tens of thousands.
> 
> The first wave of budget resin printers opened the eyes of many small businesses and consumers to the technology’s abilities and the possibilities that come along with it. However, in the beginning, availability was mostly restricted to small-sized machines. While some users may be content within the confines of a more modest resin vat, others soon realized that they needed a far larger one to accommodate the biggest and most ambitious projects they dreamt of printing.
> 
> With the growing demand and technological advancements, a flock of affordable mid to large-sized machines have been released and the supply is growing day by day.
> 
> So, we thought now would be a great time to put together our recommendations from the somewhat large-sized resin 3D printers to the extra-large printers available. We included both lower-priced 3D printers for makers at home and some higher quality options for businesses willing to invest more in a resin 3D printer.
> 
> Here are the large budget resin 3D printers worth knowing about right now.
> 
> ...
